### PR TITLE
bump sphinx to 1.7 and correct links and toctree

### PR DESCRIPTION
**Pull request by willingc**
- correcting links that fail `make linkcheck` or have new redirects
- make a couple of toctree corrections

### CODE REVIEW

File: doc/source/upgrading.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ These steps are **critical** before performing a major upgrade.
 
 ### v0.5 to v0.6
 
-See the [CHANGELOG](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/d1b2804d887033f5d0e1f5315cfe8d3c7bbf41d2/CHANGELOG.md#06---ellyse-perry---2017-01-29).
+See the [CHANGELOG](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/CHANGELOG.html#06---ellyse-perry---2017-01-29).
```
Comment (choldgraf) [inline doc/source/upgrading.md]: I don't believe that this file exists in `html` form...did you mean to link to the docs?

**Comment by consideRatio** [comment]
@willingc - anything I can help you with regarding this PR?

**Comment by choldgraf** [comment]
IMO there is a lot of valuable content in here, and I'm +1 on getting this merged sooner than later. @willingc would you like either @consideRatio or me to rebase onto master so we can merge, or do you have other edits you wanted to make to this one? (or you can rebase as well, just offering in case you're busy!)

**Comment by willingc** [comment]
Thanks @consideRatio @choldgraf. Sorry I lost this one. Thanks for resurrecting. Rebased but please double check that I didn't clobber anything by accident in the rebase.

**Comment by consideRatio** [comment]
Wieee @willingc :star2:

**Pull request merged**